ingestion/src/metadata/ingestion/source/dashboard/domodashboard/metadata.py

```diff
@@ -64,31 +69,59 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
             )
         return cls(config, metadata_config)
 
-    def get_dashboards_list(self) -> Optional[List[dict]]:
+    def get_dashboards_list(self) -> Optional[List[DomoDashboardDetails]]:
         dashboards = self.client.page_list()
         dashboard_list = []
         for dashboard in dashboards:
             dashboard_detail = self.get_page_details(page_id=dashboard["id"])
-            dashboard_list.append(dashboard_detail)
+            dashboard_list.append(
+                DomoDashboardDetails(
+                    id=dashboard_detail.id,
+                    name=dashboard_detail.name,
+                    cardIds=dashboard_detail.cardIds,
+                    description=dashboard_detail.description,
+                    owners=dashboard_detail.owners,
+                    collectionIds=dashboard_detail.collectionIds,
+                )
+            )
         return dashboard_list
 
-    def get_dashboard_name(self, dashboard: Any) -> str:
-        return dashboard["name"]
+    def get_dashboard_name(self, dashboard: DomoDashboardDetails) -> str:
+        return dashboard.name
 
-    def get_dashboard_details(self, dashboard: dict) -> dict:
+    def get_dashboard_details(self, dashboard: DomoDashboardDetails) -> dict:
         return dashboard
 
+    def get_owner_details(self, owners: List[DomoOwner]) -> Optional[EntityReference]:
```

Comment:
We have quite similar implementations of this in tableau (https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/source/dashboard/tableau/metadata.py#L237) and looker (https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/source/dashboard/looker/metadata.py#L146).

I actually like the looker one because we are caching the responses in a dictionary. Maybe it's time to centralise these functions in the dashboard_service itself?

No need to do it in this patch, but interesting to tackle next. Thanks

Reply:
Yes 👍